perf.groovy

```diff
@@ -0,0 +1,120 @@
+                    }
+                }
+
+                def configuration = 'Release'
+                def runType = isPR ? 'private' : 'rolling'
+                def benchViewName = isPR ? 'SDK private %BenchviewCommitName%' : 'SDK rolling %GIT_BRANCH_WITHOUT_ORIGIN% %GIT_COMMIT%'
+                def uploadString = '-uploadToBenchview'
+                def perfWorkingDirectory = '%WORKSPACE%\\artifacts\\${configuration}\\TestResults\\Performance'
+
+                steps {
+                    // Batch
+                    batchFile("powershell -NoProfile wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -OutFile \"${perfWorkingDirectory}\\nuget.exe\"")
+                    batchFile("if exist \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\" rmdir /s /q \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\"")
+                    batchFile("\"${perfWorkingDirectory}\\nuget.exe\" install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -OutputDirectory \"${perfWorkingDirectory}\" -Prerelease -ExcludeVersion")
+                    // Do this here to remove the origin but at the front of the branch name as this is a problem for BenchView
+                    //    we have to do it all as one statement because cmd is called each time and we lose the set environment variable
+                    batchFile("if \"%GIT_BRANCH:~0,7%\" == \"origin/\" (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH:origin/=%\") else (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH%\")\n" +
+                    "set \"BENCHVIEWNAME=${benchViewName}\"\n" +
+                    "set \"BENCHVIEWNAME=%BENCHVIEWNAME:\"=\"\"%\"\n" +
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"%BENCHVIEWNAME%\" --user-email \"dotnet-bot@microsoft.com\" --output \"${perfWorkingDirectory}\"\n" +
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch %GIT_BRANCH_WITHOUT_ORIGIN% --type ${runType} --output \"${perfWorkingDirectory}\"")
+                    batchFile("py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\" --output \"${perfWorkingDirectory}\"")
+
+			        // Build solution and run the performance tests
+				    batchFile("\"%WORKSPACE%\\build.cmd\" -perf /p:PerfIterations=10 /p:PerfOutputDirectory=\"${perfWorkingDirectory}\" /p:PerfCollectionType=stopwatch")
+
+			       //Create submission json and upload to Benchview
+				   batchFile("for /f \"tokens=*\" %%a in ('dir /b/a-d ${perfWorkingDirectory}\\*.xml') do (py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\measurement.py xunitscenario \"${perfWorkingDirectory}\\%%a\" --better desc --output \"${perfWorkingDirectory}\")")
```

Comment:
Make sure that you add --append or you will overwrite the measurements

Reply:
Done...